projects/observability/src/pages/apis/service-detail/instrumentation/service-instrumentation.component.ts

```diff
@@ -24,9 +35,24 @@ import { map } from 'rxjs/operators';
   `
 })
 export class ServiceInstrumentationComponent {
-  public serviceName$: Observable<string | undefined> = this.breadcrumbsService.breadcrumbs$.pipe(
-    map(breadcrumbs => (breadcrumbs.length > 0 ? breadcrumbs[breadcrumbs.length - 1]?.label : ''))
-  );
+  public serviceScoreSubject: BehaviorSubject<ServiceScoreResponse | undefined> = new BehaviorSubject<
```

Comment:
Any motivation for going w/ `BehaviorSubject` here? IMO, `Subject` or `Observable` should suffice.

Reply:
No particular reason, just seemed to work. Should I replace it with `Subject`?

Reply:
Yep, I think it can be an `Observable` or a `Subject`.

Reply:
Switching `BehaviorSubject` to `Subject` broke the UI. It's not rendering the data anymore.

<img width="1035" alt="Screenshot 2022-09-14 at 6 54 51 AM" src="https://user-images.githubusercontent.com/29686866/190037842-f55a3a14-f04c-45ce-8752-95a71b67a4d9.png">

```ts
// Changed this
public serviceScoreSubject: BehaviorSubject<ServiceScoreResponse | undefined> = new BehaviorSubject<
    ServiceScoreResponse | undefined
  >(undefined);

// to this
public serviceScoreSubject: Subject<ServiceScoreResponse> = new Subject<ServiceScoreResponse>();
```

Not sure I understand how `Subject` works. Is there any benefit to it over `BehaviorSubject`?

Reply:
I think that's because the UI doesn't have a check for `undefined` which the initial value of `Subject` would be. I can try to add the changes in my PR, so you aren't blocked. Will resolve and approve the PR for now.

Reply:
Probably happening because Subject's value depends on the position of subscription.

Found this: https://stackoverflow.com/a/43351340/7435656

<img width="364" alt="Screenshot 2022-09-14 at 11 52 52 AM" src="https://user-images.githubusercontent.com/29686866/190075597-0a52dca8-a643-4539-aa52-c0ee7c8b8c5c.png">

`BehaviorSubject` will output the latest value regardless of where the subscription is. To get the correct value from `Subject`, we need to subscribe before making any `next` calls. Not sure how to make `Subject` work here with async pipe.